### PR TITLE
Added optimal settings for travis

### PR DESCRIPTION
**Pull request by amilajack**

### CODE REVIEW

File: .travis.yml [hunk 1]
```diff
@@ -1,5 +1,15 @@
+sudo: false
```
Comment (1999) [inline .travis.yml]: As far as I understand TravisCI, this goes by default, isn't it?

File: .travis.yml [hunk 2]
```diff
@@ -1,5 +1,15 @@
...
+
 language: node_js
+
 node_js:
-  - "stable"
-  - "5"
-  - "4"
+  - 6
+  - 4
```
Comment (1999) [inline .travis.yml]: Why did you skip 5?

File: .travis.yml [hunk 3]
```diff
@@ -1,5 +1,15 @@
...
+
+cache:
+  directories:
+    - node_modules
```
Comment (1999) [inline .travis.yml]: Cool thing! But what if package.json dependencies change after PR is created?

File: .travis.yml [hunk 4]
```diff
@@ -1,5 +1,15 @@
...
+
+install:
+  - npm i -g npm@latest
```
Comment (1999) [inline .travis.yml]: This seems a bit unsafe to me. Each node comes with its own npm, why should we switch to latest node?

Comment (amilajack) [inline .travis.yml]: The general pattern is that travis ci versions only support LTS versions of node, which are 6 and 4. 5 and lower are dropped in newer projects and are supported by large projects that have many users.

Comment (1999) [inline .travis.yml]: Aha, that's ok. Could you update your PR so we could merge it? And what about other questions?

Comment (amilajack) [inline .travis.yml]: Sure, what should I update?

Comment (1999) [inline .travis.yml]: You didn't give answers to other questions to this PR. Seems like other changes are either unnecessary or you should tell more about them.

> sudo: false

As far as I understand TravisCI, this goes by default, isn't it?

> cache directories node_modules

Cool thing! But what if package.json dependencies change after PR is created?

> npm i -g npm@latest

This seems a bit unsafe to me. Each node comes with its own npm, why should we switch to latest node? Especially now when `mocha` is peerDependency of mocha-parallel-tests.

**Pull request closed without merging**